tests/src/JIT/opt/FastTailCall/FastTailCallCandidates.cs

```diff
@@ -0,0 +1,1150 @@
+                                          five,
+                                          six,
+                                          seven,
+                                          eight,
+                                          new StructSizeEightNotExplicit(b), 
+                                          new StructSizeEightNotExplicit(a));
+        }
+    }
+    
+    [StructLayout(LayoutKind.Explicit, Size=8, CharSet=CharSet.Ansi)]
+    public struct StructSizeThirtyTwo
+    {
+        [FieldOffset(0)]  public int a;
+        [FieldOffset(8)] public int b;
+        [FieldOffset(16)] public int c;
+        [FieldOffset(24)] public int d;
+
+        public StructSizeThirtyTwo(int a, int b, int c, int d)
+        {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+            this.d = d;
+        }
+    };
+
+    [StructLayout(LayoutKind.Explicit, Size=8, CharSet=CharSet.Ansi)]
+    public struct StructSizeTwentyFour
+    {
+        [FieldOffset(0)] public int a;
+        [FieldOffset(8)] public int b;
+        [FieldOffset(16)] public int c;
+
+        public StructSizeTwentyFour(int a, int b, int c)
+        {
+            this.a = a;
+            this.b = b;
+            this.c = c;
+        }
+    }
+
+    /// <summary>
+    /// Decision to fast tail call
+    /// </summary>
+    /// <remarks>
+    ///
+    /// The callee has one stack argument of size 8
+    ///
+    /// The callee should not fast tail call
+    ///
+    /// Return 100 is a pass.
+    /// Return 105 is a failure.
+    ///
+    /// </remarks>
+    public static int StackBasedCallee(int a, int b, StructSizeThirtyTwo sstt)
+    {
+        int count = 0;
+        for (int i = 0; i < sstt.a; ++i)
+        {
+            if (i % 10 == 0)
+            {
+                ++count;
+            }
+        }
+
+        if (count == 160)
+        {
+            return 100;
+        }
+
+        else
+        {
+            return 105;
+        }
+    }
+
+    /// <summary>
+    /// Decision to fast tail call
+    /// </summary>
+    /// <remarks>
+    ///
+    /// On x64 linux this will not fastTailCall.
+    ///
+    /// The caller has one stack argument of size 24
+    /// The callee has one stack argument of size 32
+    ///
+    /// On Arm64 this will fastTailCall
+    ///
+    /// Both caller and callee have two register args.
+    ///
+    /// Return 100 is a pass.
+    /// Return 105 is a failure.
+    ///
+    /// </remarks>
+    public static int StackBasedCaller(int i, StructSizeTwentyFour sstf)
+    {
+        if (i % 2 == 0)
+        {
+            int a = i * 100;
+            int b = i + 1100;
+            return StackBasedCallee(a, b, new StructSizeThirtyTwo(a, b, b, a));
+        }
+        else
+        {
+            int b = i + 829;
+            int a = i + 16;
+            return StackBasedCallee(b, a, new StructSizeThirtyTwo(b, a, a, b));
+        }
+    }
+
+    public static int Main()
+    {
+        return Tester(1);
+    }
```

Comment:
This is a fair comment. Most of these test cases just assert before the canFastTailCall changes. So a pass at the time of writing them was not an assert and manually checking the decisions printed by the logging. To entirely validate each test would probably need to check the output and see if a fast tail call happened or not. Note that this is complicated by the fact that we currently make the wrong decisions due to implementation limitations and bugs. So the test will have to be changed as those changes are made.